encoding.go

```diff
@@ -0,0 +1,89 @@
+package verkle
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+const ErrInvalidNodeEncoding = "invalid node encoding"
+
+func ParseNode(serialized []byte, tc *TreeConfig) (VerkleNode, error) {
```

Comment:
I suggest changing the signature of this function to `(serialized []byte, width int)` because then you can pass the result of calling `GetTreeConfig(width)` as a parameter to `createInternalNode`. This way, `TreeConfig` is that much closer to not being exported outside the `verkle` package.